.github/workflows/publish.yml

```diff
@@ -0,0 +1,25 @@
+name: Publish to Maven Central
+on:
+  release:
+    types:
+      - created
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 11
+          cache: 'gradle'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+      - name: Publish with Gradle
```

Comment:
Not sure I understand the lifecycle of GH releases fully, but after this step, don't you have to call `gh release publish` or something like that?